### PR TITLE
In the barchart show term2 value in list samples if there is overlay

### PR DESCRIPTION
**Pull request by airenzp**
## Description

Show term2 value in list samples if there is overlay

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Comment by karishma-gangwani** [comment]
Thanks for the fix @airenzp. If the term is a numeric term then the value needs to be shown in the second column regardless of t1 or t2. because term 1 can be numeric while overlay could be categorical. maybe we can show both the columns for term 1 and term 2 when 'list samples' is clicked?

**Review by creilly8 — APPROVED** [review]
Works well, thanks.

**Comment by karishma-gangwani** [comment]
Thanks @airenzp. I see an error in [ash](http://localhost:3000/?noheader=1&massnative=hg38,ASH) ds:

in term 1 add 'tp53' from mutation tab and term2 'age at diagnosis' from dictionary variables. client side shows an error when samples are listed from the 'list samples' option for any of the groups. can you replicate and see if that's real? This might need to expand to other term types. 

image

**Comment by airenzp** [comment]
> Thanks @airenzp. I see an error in [ash](http://localhost:3000/?noheader=1&massnative=hg38,ASH) ds:
> 
> in term 1 add 'tp53' from mutation tab and term2 'age at diagnosis' from dictionary variables. client side shows an error when samples are listed from the 'list samples' option for any of the groups. can you replicate and see if that's real? This might need to expand to other term types.
> 
> image

Should be fixed now

**Review by karishma-gangwani — DISMISSED** [review]
thanks for the fixes @airenzp

**Review by creilly8 — DISMISSED** [review]

File: shared/utils/src/terms.js [hunk 1]
```diff
@@ -153,6 +153,7 @@ export function getBin(lst, value) {
 
 export function getSampleType(term, ds) {
```
Comment (xzhou82) [inline shared/utils/src/terms.js:154]: purpose of this function is not obvious. should add comment

File: shared/utils/src/terms.js [hunk 2]
```diff
@@ -153,6 +153,7 @@ export function getBin(lst, value) {
...
 	if (!term) return null
+	if (term.type && isNonDictionaryType(term.type)) return DEFAULT_SAMPLE_TYPE
```
Comment (xzhou82) [inline shared/utils/src/terms.js:156]: please explain logic in comment

**Comment by creilly8** [comment]
Hey @airenzp. I'm seeing this error from http://localhost:3000/?noheader=1&massnative=hg38,ASH. 
Screenshot 2024-11-19 at 9 21 42 AM
Browser console: 
```
Uncaught TypeError: Cannot read properties of undefined (reading 'term1Label')
    at Object.mouseover (barchart.events.js:78:40) // this line: const term1Label = d.groupPvalues.term1Label
```
I refreshed my ASH files and restarted my instance. I still see the same error.

**Comment by airenzp** [comment]
This is not related to this PR, I actually noticed this type of errors when adding the sample values and added question marks in this line `const pvalue = d.groupPvalues?.term2tests?.find(x => x.term2id === d.dataId).pvalue` to fix it. Lets make an issue, I could look into it later but am not the one who coded this logic, so maybe someone else can help faster. 
This PR should be ready to merge actually @xzhou82.

**Comment by xzhou82** [comment]
server err exists on master for ASH geneVariant+term overlay. maybe related to server require assayAvailability where it should be optional

**Review by xzhou82 — APPROVED** [review]

**Pull request merged**